src/include/type/varlen_pool.h

```diff
@@ -96,31 +84,11 @@ class VarlenPool {
   // Get the maximum size of this pool.
   uint64_t GetMaximumPoolSize() const;
 
-  // Get the empty buffer count for a given empty buffer list id
-  // Return -1 if list_id is out of bound
-  int GetEmptyCountByListId(size_t list_id) const;
-
-  static size_t GetRefCountSize() {
-    return sizeof(std::atomic<int64_t>);
-  }
-
-private:
-  inline std::atomic<int64_t> *GetRefCntPtr(void *addr) {
-    char *ref_cnt_addr = ((char *) addr) - sizeof(std::atomic<int64_t>);
-    return reinterpret_cast<std::atomic<int64_t> *>(ref_cnt_addr);
-  }
-
-private:
+public:
   // All these fields are implementation specific.
   // This class must be thread-safe, very very fast and provide some form of
   // compaction or garbage-collection.
 
-  // Internal memory allocation
-  void *AllocateHelper(size_t size);
-
-  // Internal memory deallocation

```

Comment:
I don't think we need to remove all the old varlen code including the ref count optimization. A better way would be to abstract an interface, keep the old code and implement a new "ephemeral" pool.